cayenne/src/main/java/org/apache/cayenne/access/translator/select/DistinctStage.java

```diff
@@ -70,7 +70,7 @@ public void perform(TranslatorContext context) {
         context.getSelectBuilder().distinct();
     }
 
-    private boolean hasToManyJoin(TranslatorContext context) {
+    static boolean hasToManyJoin(TranslatorContext context) {
```

Comment:
We may move this logic to the `TableTree` class to keep stages as separeted as possible.

Reply:
Done